app/models/package_manager/maven.rb

```diff
@@ -17,27 +18,52 @@ class Maven < Base
       "http://www.eclipse.org/org/documents/edl-v10" => "Eclipse Distribution License (EDL), Version 1.0",
     }.freeze
 
+    PROVIDER_MAP = {
+      "SpringLibs" => SpringLibs,
+      "Maven" => MavenCentral,
+    }.freeze
+
     def self.package_link(project, version = nil)
+      if version
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
+          return PROVIDER_MAP[repository_source].package_link(project, version)
+        end
+      end
+
       MavenUrl.from_name(project.name, repository_base).search(version)
     end
 
     def self.download_url(name, version = nil)
+      if version
+        project = Project.find_by(name: name, platform: "Maven")
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
+          return PROVIDER_MAP[repository_source].download_url(name, version)
+        end
+      end
+
       MavenUrl.from_name(name, repository_base).jar(version)
     end
 
     def self.check_status_url(project)
-      MavenUrl.from_name(project.name, repository_base).base
-    end
+      sources = project.versions.flat_map(&:repository_sources).compact.uniq
```

Comment:
Should we just go for the last version?

Reply:
By publish date or by created_at date? I'm wondering if we can still do some sort of priority on sources. Maybe setting that in the Maven source map and pick the highest priority source from the list in `versions`.

Reply:
Not even really sure what a check status url is...

Reply:
This is the URL that is used to check if the project is deprecated or not. So this just needs to not 404.

Reply:
https://github.com/librariesio/libraries.io/blob/005d6de3e86c2354f78183dccc690a004d60fcf9/app/models/project.rb#L520